algoliasearch-common/src/main/java/com/algolia/search/inputs/analytics/ABTest.java

```diff
@@ -0,0 +1,38 @@
+package com.algolia.search.inputs.analytics;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public class ABTest {
+  // Properties available at construction time
+  public String name;
+  public List<Variant> variants;
+  public ZonedDateTime endAt;
+
+  // Properties set by the AB Testing API
+  @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
```

Comment:
why all the `@JsonProperty(access = JsonProperty.Access.WRITE_ONLY)` ?

Reply:
This is one way to make Jackson **only** deserialize a property (i.e. force it not to serialize those fields). In our case, it is useful because most of the fields are read-only and produced by the Analytics API upon read. When adding/editing tests, the user can only write the top fields that are the only ones accessible as RW.

I'll remove it anyway following your recommendation to use getter/setter rather than public attributes. Because if I only provide the setter, it has the same effect as this annotation.

For the record, I've found all those informations here:
http://www.baeldung.com/jackson-field-serializable-deserializable-or-not.